protocol/xmrmaker/checks.go

```diff
@@ -80,3 +84,90 @@ func checkContractSwapID(msg *message.NotifyETHLocked) error {
 
 	return nil
 }
+
+// checkContract checks the contract's balance and Claim/Refund keys.
+// if the balance doesn't match what we're expecting to receive, or the public keys in the contract
+// aren't what we expect, we error and abort the swap.
+func (s *swapState) checkContract(txHash ethcommon.Hash) error {
```

Comment:
this hasn't been modified, I just copy pasted it from swap_state.go since that file was getting long